### PR TITLE
Modify S3335: Change text to LayC

### PR DESCRIPTION
**Pull request by gaetan-ferry-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S3335/php/rule.adoc [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
 
-The ``++cgi.force_redirect++`` _php.ini_ configuration is on by default, and it prevents unauthenticated access to scripts when PHP is running as a CGI. Unfortunately, it must be disabled on IIS, OmniHTTPD and Xitami, but in all other cases it should be on.
+Pre-processing on the server side is often required to check users
+authentication when working in CGI mode. Those preliminary actions can also
+position diverse configuration parameters that are necessary for the CGI script
+to work properly.
+
+If this configuration is not set up before access, the CGI might behave
+unexpectedly, which can have various consequences. Mainly, authentication or
+authorization are common threats in those cases.
```
Comment (loris-s-sonarsource) [inline rules/S3335/php/rule.adoc:17]: Here, "auth or authz are common threats" does not feel right. Also, I would suggest adding this paragraph into a "whats the impact" section and maybe talk about authentication or authorization to young developers who would not understand what this means in practice. But really as you feel it

Comment (gaetan-ferry-sonarsource) [reply rules/S3335/php/rule.adoc:17]: :confounded:

File: rules/S3335/php/rule.adoc [hunk 2]
```diff
@@ -1,27 +1,61 @@
...
 
+The `cgi.force_redirect` is not supported by all web servers. For example,
+Microsoft IIS web server is unable to differentiate an internally redirected
+request from a normal one.
+
+If using such a server, the `cgi.force_redirect` parameter will have to be
+disabled for the CGI scripts to work properly. In that case, it is important to
+ensure the CGI behavior is aware of the security threat.
+
+== Resources
```
[the remark this answers is not in the document]
Comment (gaetan-ferry-sonarsource) [reply rules/S3335/php/rule.adoc:55]: :confounded:

File: rules/S3335/php/rule.adoc [hunk 3]
```diff
@@ -1,27 +1,73 @@
...
+
+=== Pitfalls
+
+The `cgi.force_redirect` is not supported by all web servers. For example,
+Microsoft IIS web server is unable to differentiate an internally redirected
+request from a normal one.
+
+If using such a server, the `cgi.force_redirect` parameter will have to be
+disabled for the CGI scripts to work properly. In that case, it is important to
+ensure the CGI behavior is aware of the security threat.
```
Comment (loris-s-sonarsource) [inline rules/S3335/php/rule.adoc:62]: "If using" feels a little string, I would suggest "while" or "if you use"

```suggestion
While using such a server, the `cgi.force_redirect` parameter will have to be
disabled for the CGI scripts to work properly. In that case, it is important to
ensure the CGI behavior is aware of the security threat.
```

**Review by loris-s-sonarsource — APPROVED** [review]
LGTM !

**Pull request merged**